src/components/tools/common.ts

```diff
@@ -0,0 +1,75 @@
+import { AnnotationToolStore } from '@/src/store/tools/useAnnotationTool';
+import { Maybe } from '@/src/types';
+import { AnnotationTool } from '@/src/types/annotation-tool';
+import { vtkAnnotationToolWidget } from '@/src/vtk/ToolWidgetUtils/utils';
+import vtkAbstractWidgetFactory from '@kitware/vtk.js/Widgets/Core/AbstractWidgetFactory';
+import vtkWidgetState from '@kitware/vtk.js/Widgets/Core/WidgetState';
+import { VNode } from 'vue';
+
+export interface WidgetComponentMeta<
+  ToolID extends string,
+  ToolStore extends AnnotationToolStore<ToolID>,
+  WidgetState extends vtkWidgetState,
+  WidgetFactory extends vtkAbstractWidgetFactory,
+  ViewWidget extends vtkAnnotationToolWidget,
+  SyncedState,
+  InitState
+> {
+  /**
+   * The name of the component.
+   */
+  name: string;
+
+  /**
+   * The associated tool store.
+   */
+  useToolStore: () => ToolStore;
+
+  /**
+   * Construct a new standalone widget state.
+   * Used by the placing widget component.
+   */
+  createStandaloneState: () => WidgetState;
+
+  /**
+   * Construct a new store-backed state.
+   */
+  createStoreBackedState: (id: ToolID, store: ToolStore) => WidgetState;
+
+  /**
+   * Construct a widget factory.
+   */
+  createWidgetFactory: (widgetState: WidgetState) => WidgetFactory;
+
+  /**
+   * A composable that syncs the widget factory's state to a reactive object.
+   */
+  useSyncedState: (widgetFactory: WidgetFactory) => SyncedState;
```

Comment:
This is needed because PlacingWidget does not pull from a store and we want a normalized representation of state for Placing and NonPlacing components?  We do pass the `tool` to store backed NonPlacing, so passing duplicate info to NonPlacing components?

Reply:
We are certainly passing some duplicate info, but the synced state captures some ephemeral widget state that the store might not capture. For example, the polygon tool currently stores a `movePoint`. I consider that to be ephemeral state that doesn't need to be stored, only propagated from the widget itself.